File: java/cdk8s-plus-elasticsearch-query/src/main/java/imports/co/elastic/k8s/elasticsearch/ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable.java
package imports.co.elastic.k8s.elasticsearch;

/**
 * An eviction is allowed if at most "maxUnavailable" pods selected by "selector" are unavailable after the eviction, i.e. even in absence of the evicted pod. For example, one can prevent all voluntary evictions by specifying 0. This is a mutually exclusive setting with "minAvailable".
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.79.0 (build b22f628)", date = "2023-03-31T16:02:14.061Z")
@software.amazon.jsii.Jsii(module = imports.co.elastic.k8s.elasticsearch.$Module.class, fqn = "coelastick8selasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable")
public class ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable extends software.amazon.jsii.JsiiObject {

    protected ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    public static @org.jetbrains.annotations.NotNull imports.co.elastic.k8s.elasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable fromNumber(final @org.jetbrains.annotations.NotNull java.lang.Number value) {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(imports.co.elastic.k8s.elasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable.class, "fromNumber", software.amazon.jsii.NativeType.forClass(imports.co.elastic.k8s.elasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable.class), new Object[] { java.util.Objects.requireNonNull(value, "value is required") });
    }

    public static @org.jetbrains.annotations.NotNull imports.co.elastic.k8s.elasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable fromString(final @org.jetbrains.annotations.NotNull java.lang.String value) {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(imports.co.elastic.k8s.elasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable.class, "fromString", software.amazon.jsii.NativeType.forClass(imports.co.elastic.k8s.elasticsearch.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable.class), new Object[] { java.util.Objects.requireNonNull(value, "value is required") });
    }

    public @org.jetbrains.annotations.NotNull java.lang.Object getValue() {
        return software.amazon.jsii.Kernel.get(this, "value", software.amazon.jsii.NativeType.forClass(java.lang.Object.class));
    }
}

File: go/cdk8s-plus-elasticsearch-query/imports/elasticsearchk8selasticco/ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable.go
// elasticsearchk8selasticco
package elasticsearchk8selasticco

import (
	_init_ "example.com/cdk8s-plus-elasticsearch-query/imports/elasticsearchk8selasticco/jsii"
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"
)

// An eviction is allowed if at most "maxUnavailable" pods selected by "selector" are unavailable after the eviction, i.e. even in absence of the evicted pod. For example, one can prevent all voluntary evictions by specifying 0. This is a mutually exclusive setting with "minAvailable".
type ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable interface {
	Value() interface{}
}

// The jsii proxy struct for ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable
type jsiiProxy_ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable struct {
	_ byte // padding
}

func (j *jsiiProxy_ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable) Value() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"value",
		&returns,
	)
	return returns
}


func ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable_FromNumber(value *float64) ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable {
	_init_.Initialize()

	if err := validateElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable_FromNumberParameters(value); err != nil {
		panic(err)
	}
	var returns ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable

	_jsii_.StaticInvoke(
		"elasticsearchk8selasticco.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable",
		"fromNumber",
		[]interface{}{value},
		&returns,
	)

	return returns
}

func ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable_FromString(value *string) ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable {
	_init_.Initialize()

	if err := validateElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable_FromStringParameters(value); err != nil {
		panic(err)
	}
	var returns ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable

	_jsii_.StaticInvoke(
		"elasticsearchk8selasticco.ElasticsearchSpecPodDisruptionBudgetSpecMaxUnavailable",
		"fromString",
		[]interface{}{value},
		&returns,
	)

	return returns
}

